File: problems/1133/solution.cpp
class Solution {
public:
    int largestUniqueNumber(vector<int>& A) {
        int ans = -1;
        unordered_map<int, int> m;
        
        for (const int& num: A) {
            m[num]++;
        }
        
        for (auto [num, freq]: m) {
            if (freq == 1) {
                ans = max(ans, num);
            }
        }
        
        return ans;
    }
};

File: problems/1133/solution.go
package main
func largestUniqueNumber(A []int) int {
    if len(A) == 1 {
        return A[0]
    }
    
    sort.Ints(A)
    
    for i := len(A) - 1; i >= 0; i-- {
        if (i == len(A) - 1 || A[i] != A[i+1]) && (i == 0 || A[i] != A[i-1]) {
            return A[i]
        }
    }
    
    return -1
}